org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/SaveActionHandler.java

```diff
@@ -69,7 +72,11 @@ public List<TextEdit> willSaveWaitUntil(WillSaveTextDocumentParams params, IProg
 		}
 
 		LinkedHashSet<String> cleanUpIds = new LinkedHashSet<>();
-		List<String> lspCleanups = preferences.getCleanUpActionsOnSave();
+
+		List<String> lspCleanups = Collections.emptyList();
+		if (preferences.getCleanUpActionsOnSaveEnabled()) {
```

Comment:
Also, `testNoConflictBetweenLSPAndJDTUI` fails because of this PR. Just needs to be adjusted to new requirement.